src/parser/parser.cpp

```diff
@@ -166,8 +166,56 @@ bool Parser::WriteParseTree(const std::string &filepath) {
     ofs << parse_tree_.dump(INDENTATION) << '\n';
     return true;
   }
-
   return false;
 }
 
+/**
+ * Supportive function for Parser::FormattedJSON
+ * @param value
+ * @return { text: { name: "value" } }
+ */
+json GetTextNode(const std::string &value) {
+  json j;
+  j["text"]["name"] = value;
+  return j;
+}
+
+/**
+ * Supportive function for Parser::FormattedJSON
+ * @param value
+ * @return the current key of the JSON head
+ */
+std::string GetHeadKey(const json &j) {
+  for (auto it = j.begin(); it != j.end();) {
+    return it.key();
+  }
+  return "";
+}
+
+/**
+ * Utility recursive function for Parser::FormattedJSON
+ * @param body, a json
+ * @returns Treant.js formatted JSON
+ */
+json RecRunner(json main, std::string key = "") {
```

Comment:
Make const &

Reply:
Wouldn't make much of a difference from a memory perspective.
Inside the body, we have to create a copy of this constant to handle base cases.